tests/unit/lms/db/_bulk_action_test.py

```diff
@@ -57,78 +46,37 @@ def test_upsert(self, db_session):
 
         assert isinstance(result, CursorResult)
 
-        rows = list(db_session.query(self.TableWithBulkUpsert))
-
-        assert (
-            rows
-            == Any.iterable.containing(
-                [
-                    Any.instance_of(self.TableWithBulkUpsert).with_attrs(expected)
-                    for expected in [
-                        {"id": 1, "name": "update_old", "other": "pre_1"},
-                        {"id": 2, "name": "pre_existing_2", "other": "pre_2"},
-                        {"id": 3, "name": "create_with_id", "other": "post_3"},
-                        {"id": 4, "name": "over_block_size", "other": "post_4"},
-                    ]
-                ]
-            ).only()
+        self.assert_has_rows(
+            db_session,
+            {"id": 1, "name": "update_old", "other": "pre_1"},
+            {"id": 2, "name": "pre_existing_2", "other": "pre_2"},
+            {"id": 3, "name": "create_with_id", "other": "post_3"},
+            {"id": 4, "name": "over_block_size", "other": "post_4"},
         )
 
     def test_upsert_does_nothing_if_given_an_empty_list_of_values(self, db_session):
         assert BulkAction(db_session).upsert(self.TableWithBulkUpsert, []) == []
 
-    def test_upsert_with_onupdate_columns(self, db_session):
+    @pytest.mark.parametrize("column", ("scalar", "callable", "sql", "default"))
+    @pytest.mark.usefixtures("with_upsert_trigger_onupdate")
+    def test_upsert_with_onupdate_columns(self, db_session, column):
         db_session.add_all(
             [
-                self.TableWithBulkUpsertOnUpdate(
-                    id=1, name="existing", scalar=0, callable=0, sql=0, default=0
-                ),
-                self.TableWithBulkUpsertOnUpdate(
-                    id=2, name="existing", scalar=1, callable=1, sql=1, default=1
-                ),
+                self.TableWithBulkUpsert(id=1, name="pre_existing_1", **{column: 0}),
+                self.TableWithBulkUpsert(id=2, name="pre_existing_2", **{column: 1}),
```

Comment:
I followed the naming as above here with `pre_...` so the two can be seen apart by name.